Modules/Microsoft365DSC/DSCResources/MSFT_AADSecurityDefaults/MSFT_AADSecurityDefaults.psm1

```diff
@@ -0,0 +1,362 @@
+            ApplicationId         = $ApplicationId
+            TenantId              = $TenantId
+            ApplicationSecret     = $ApplicationSecret
+            CertificateThumbprint = $CertificateThumbprint
+            Credential            = $Credential
+        }
+
+        Write-Verbose -Message "Get-TargetResource Result: `n $(Convert-M365DscHashtableToString -Hashtable $result)"
+        return $result
+    }
+    catch
+    {
+        try
+        {
+            Write-Verbose -Message $_
+            $tenantIdValue = ""
+            if (-not [System.String]::IsNullOrEmpty($TenantId))
+            {
+                $tenantIdValue = $TenantId
+            }
+            Add-M365DSCEvent -Message $_ -EntryType 'Error' `
+                -EventID 1 -Source $($MyInvocation.MyCommand.Source) `
+                -TenantId $tenantIdValue
+        }
+        catch
+        {
+            Write-Verbose -Message $_
+        }
+        return $nullReturn
+    }
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        [ValidateSet('Yes')]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.String]
+        $DisplayName,
+
+        [Parameter()]
+        [System.String]
+        $Description,
+
+        [Parameter()]
+        [System.Boolean]
+        $IsEnabled,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationSecret,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint
+    )
+
+    Write-Verbose -Message "Setting configuration for Azure AD Security Defaults"
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace "MSFT_", ""
+    $CommandName  = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    if ($CurrentValues.DisplayName -ne $DisplayName)
+    {
+        Write-Warning -Message "The DisplayName property for the AADSecurityDefaults resource is read-only and cannot be changed."
+    }
+
+    if ($CurrentValues.Description -ne $Description)
+    {
+        Write-Warning -Message "The Description property for the AADSecurityDefaults resource is read-only and cannot be changed."
```

Comment:
Why add this as a parameter if it is read-only?

Reply:
For reporting purposes, mostly. I know this is currently set as Single Instance, but my guts feeling tells me that there may soon be a way to define more then 1 security defaults. I am willing to change this is you believe it will add confusion to the users. I am not a fan of exposing properties that cannot be changed, but otherwise this resource because a single boolean resource

Reply:
Then maybe not use Write-Warning, but Write-Verbose instead and add an "Informational Only" text or something.

To me, showing a warning message triggers a "Hey, something is wrong and I need to do something" responds. Where in this case, that is not necessary.